internal/bminventory/inventory.go

```diff
@@ -2125,61 +2122,63 @@ func (b *bareMetalInventory) updateNetworkParams(params installer.UpdateClusterP
 		updates["service_network_cidr"] = serviceCidr
 	}
 
-	if params.ClusterUpdateParams.UserManagedNetworking != nil && swag.BoolValue(params.ClusterUpdateParams.UserManagedNetworking) != userManagedNetworking {
-		userManagedNetworking = swag.BoolValue(params.ClusterUpdateParams.UserManagedNetworking)
-		updates["user_managed_networking"] = userManagedNetworking
-		machineCidr = ""
-	}
-	if userManagedNetworking && !common.IsSingleNodeCluster(cluster) {
-		err, vipDhcpAllocation = setCommonUserNetworkManagedParams(params.ClusterUpdateParams, common.IsSingleNodeCluster(cluster), machineCidr, updates, log)
-		if err != nil {
-			return err
-		}
-	}
-
-	if params.ClusterUpdateParams.VipDhcpAllocation != nil && swag.BoolValue(params.ClusterUpdateParams.VipDhcpAllocation) != vipDhcpAllocation {
-		vipDhcpAllocation = swag.BoolValue(params.ClusterUpdateParams.VipDhcpAllocation)
-		updates["vip_dhcp_allocation"] = vipDhcpAllocation
-		updates["api_vip"] = ""
-		updates["ingress_vip"] = ""
-		machineCidr = ""
-		setMachineNetworkCIDRForUpdate(updates, machineCidr)
-	}
 	if params.ClusterUpdateParams.NetworkType != nil && params.ClusterUpdateParams.NetworkType != cluster.NetworkType {
 		b.setUsage(true, usage.NetworkTypeSelectionUsage, &map[string]interface{}{
 			"network_type": params.ClusterUpdateParams.NetworkType}, usages)
 
 		updates["network_type"] = swag.StringValue(params.ClusterUpdateParams.NetworkType)
 	}
 
-	if !userManagedNetworking {
-		if vipDhcpAllocation {
-			err = b.updateDhcpNetworkParams(updates, params, log, &machineCidr)
-		} else {
-			err = b.updateNonDhcpNetworkParams(updates, cluster, params, log, &machineCidr, interactivity)
+	if params.ClusterUpdateParams.UserManagedNetworking != nil && swag.BoolValue(params.ClusterUpdateParams.UserManagedNetworking) != userManagedNetworking {
+		// User network mode has changed
+		userManagedNetworking = swag.BoolValue(params.ClusterUpdateParams.UserManagedNetworking)
+		updates["user_managed_networking"] = userManagedNetworking
+		primaryMachineNetworkCIDR = ""
+	}
+
+	if userManagedNetworking {
+		if params.ClusterUpdateParams.MachineNetworkCidr != nil {
```

Comment:
i think it can cause not relevant error in case machine cidr will fail on https://github.com/openshift/assisted-service/pull/2307/files#diff-747a429a0e12307ef65d754c83b6898ce3858b621005e1e7af6e420da1f1657cR2142 but it should not even check it in case userManagedNetworking is set and it is not SNO 
I think https://github.com/openshift/assisted-service/pull/2307/files#diff-747a429a0e12307ef65d754c83b6898ce3858b621005e1e7af6e420da1f1657cR2148
should be before this line as it validates all relevant cases

Reply:
Updated.